### PR TITLE
Eobs 761 server error message displayed for refusal following a partial observation

### PR DESCRIPTION
**Pull request by moderatemisbehaviour**
- Frequency is now correctly set on new observations created after partials, instead of letting it default to 15.
- Added docstrings.

### CODE REVIEW

File: nh_observations/frequencies.py [hunk 1]
```diff
@@ -44,18 +44,13 @@
...
 }
 
-# To summarise the mappings below, frequencies after refusals are capped at
-# 24 hours.
-#
-# Could have just made a simple function but decided to write them out long
-# hand as they may change in the future. Return the tuple with the label rather
-# than just the time so it can be useful in more scenarios.
-PATIENT_REFUSAL_ADJUSTMENTS = {
+# Frequency tuples organised by risk and frequency in minutes.
```
Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: Add tech debt to make this a model and only have particular parts in their relevant modules.

File: nh_observations/frequencies.py [hunk 2]
```diff
@@ -67,23 +62,32 @@
...
     'Obs Restart': EVERY_HOUR
 }
 
+# To summarise the mappings below, frequencies after refusals are capped at
+# 24 hours.
+#
+# Could have just made a simple function but decided to write them out long
+# hand as they may change in the future. Return the tuple with the label rather
+# than just the time so it can be useful in more scenarios.
+PATIENT_REFUSAL_ADJUSTMENTS = dict.copy(FREQUENCIES_BY_RISK)
```
Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: Is this a shallow copy or a deep copy? Make sure it's a deep copy.

File: nh_observations/frequencies.py (hunk 1 above)
Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: Added.

File: nh_observations/frequencies.py (hunk 2 above)
Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: It was in fact a shallow copy. Good catch @Gimpneek, would've almost definitely caused bugs in the future.

Comment (moderatemisbehaviour) [inline nh_observations/frequencies.py]: Changing to `copy.deepcopy`.

**Comment by Gimpneek** [comment]
You get a :shipit:! You get a :shipit:! You all get a :shipit:

**Comment by moderatemisbehaviour** [comment]
yaaay :shipit:s for everyone 🎄

**Pull request merged**